ui/repetition.reel/repetition.js

```diff
@@ -700,6 +719,62 @@ var Repetition = exports.Repetition = Component.specialize(/** @lends Repetition
     // Implementation:
     // ----
 
+    _selection: {
+        value: null
+    },
+
+    _cancelSelectionRangeChangeListener: {
+        value: null
+    },
+
+    selection: {
+        get: function () {
+            return this._selection;
+        },
+        set: function (value) {
+            if (this._selection !== value) {
+                this._selection = value;
+                if (this.contentController && this.contentController.selection !== value) {
+                    this.contentController.selection = value;
+                }
+                if (this._cancelSelectionRangeChangeListener) {
+                    this._cancelSelectionRangeChangeListener();
+                }
+                if (value) {
+                    this._cancelSelectionRangeChangeListener = (
+                        value.addRangeChangeListener(this, "selection")
+                    );
+                    this.handleSelectionRangeChange(value, []);
+                } else {
+                    this._cancelSelectionRangeChangeListener = null;
+                }
+            }
+        }
+    },
+
+    handleSelectionRangeChange: {
```

Comment:
Is it really worth creating a new Map (garbage) every time the selection changes, to find the iteration object, compared to doing a loop ourselves on iterations to find this.iterations[i].object === aObjet in add/remove?